### PR TITLE
Capture canvas screenshot (for camera perspective memory bar)

### PR DESCRIPTION
**Pull request by benptc**
Goes with: https://github.com/ptcrealitylab/vuforia-spatial-remote-operator-addon/pull/202, to enable getting a thumbnail image of the scene. Just doing canvas.toDataURL at arbitrary moments elsewhere in the code may result in an empty image in between canvas draw calls.

camera-position-memory-bar-v5

### CODE REVIEW

File: src/gui/threejsScene.js [hunk 1]
```diff
@@ -1244,6 +1247,52 @@ import {setMatrixFromArray} from "./scene/utils.js";
         return new THREE.Vector3(x, y, z);
     }
 
+    let pendingCapture = null;
+
+    function capture(outputWidth, outputHeight, compressionOptions) {
```
Comment (hobinjk-ptc) [inline src/gui/threejsScene.js:1252]: Should document this function's role and parameters

File: src/splatting/Splatting.js [hunk 2]
```diff
@@ -1176,6 +1177,52 @@ async function main(initialFilePath) {
     }
 }
 
+let pendingCapture = null;
+
+function capture(outputWidth, outputHeight, compressionOptions) {
```
Comment (hobinjk-ptc) [inline src/splatting/Splatting.js:1182]: See previous comment, also not ideal to have this level of repeated code but I guess it's okay for a little feature like this

**Review by hobinjk-ptc — APPROVED** [review]
Excited for the return of the memory bar! 😆

File: src/splatting/Splatting.js (hunk 2 above)
Comment (Steve-KX-RL) [reply src/splatting/Splatting.js:1182]: 👆agreed. Maybe if we have time to revisit this later? Since it's a small feature.

**Review by Steve-KX-RL — APPROVED** [review]
LGTM! Thx for this feature!

**Review by dangond-ptc — APPROVED** [review]
LGTM!

File: src/splatting/Splatting.js (hunk 2 above)
Comment (benptc) [reply src/splatting/Splatting.js:1182]: I'll see if there's a sensible way for me to refactor this before I merge the PR... 🤔

**Pull request merged**